src/module_2/Module_2_EDA_Ex1.md

```diff
@@ -0,0 +1,3152 @@
+     2   user_nuts1             4932 non-null   object        
+     3   first_ordered_at       4983 non-null   datetime64[us]
+     4   customer_cohort_month  4983 non-null   datetime64[us]
+     5   count_people           325 non-null    float64       
+     6   count_adults           4983 non-null   float64       
+     7   count_children         325 non-null    float64       
+     8   count_babies           325 non-null    float64       
+     9   count_pets             325 non-null    float64       
+     10  count_regulars         4983 non-null   float64       
+    dtypes: datetime64[us](2), float64(6), object(3)
+    memory usage: 428.4+ KB
+
+
+## 2. Understanding the problem
+Understand the problem at hand through data interrogation and hypothesis testing.
+
+Gather all the hypothesis and insights you have tested and found.
+
+
+Data interrogation:
+* How many products does the company sell
+* How many products are abandoned
+* How many regular products are sold 
+* What are the best selling products
+* Which are the top clients
+* What is the profile of the clients
+* Are our clients loyal, or do they only have one order?
+* Do users define their regular products?
+* What kind of products do users define as their regulars
+
+**a. How many products does the company sell each day**
+
+First, I calculate the number of products bought in each order:
+
+
+```python
+data['product_order_count'] = data['ordered_items'].apply(len)
+```
+
+    /tmp/ipykernel_154450/1249169935.py:1: SettingWithCopyWarning: 
+    A value is trying to be set on a copy of a slice from a DataFrame.
+    Try using .loc[row_indexer,col_indexer] = value instead
+    
+    See the caveats in the documentation: https://pandas.pydata.org/pandas-docs/stable/user_guide/indexing.html#returning-a-view-versus-a-copy
+      data['product_order_count'] = data['ordered_items'].apply(len)
+
+
+Now, plot an histogram with the number of sold products. For clarity, I do a different plot for each year and plot the monthly sales.
+
+
+```python
+# Aggregate counts by order year and month
+grouped_data = data.groupby([data['order_date'].dt.year, data['order_date'].dt.month])
+monthly_sales = grouped_data['product_order_count'].sum()
+
+for year in monthly_sales.index.get_level_values(0).unique():
+    monthly_sales.loc[year].plot(kind='bar')
+    plt.title(f'Number of Products Sold Each Month in the year {year}')
+    plt.xlabel('Order Date')
+    plt.ylabel('Products Sold')
+    plt.xticks(ticks=range(len(monthly_sales.loc[year])), labels=[f'{year}-{month:02d}' for month in monthly_sales.loc[year].index], rotation=45)
+    plt.show()
+```
+
+
+    
+![png](Module_2_EDA_Ex1_files/Module_2_EDA_Ex1_98_0.png)
+    
+
+
+
+    
+![png](Module_2_EDA_Ex1_files/Module_2_EDA_Ex1_98_1.png)
+    
+
+
+
+    
+![png](Module_2_EDA_Ex1_files/Module_2_EDA_Ex1_98_2.png)
+    
+
+
+To get a more general overview, plot all the years together:
```

Comment:
Exactly, above plots have different scales and are misleading. I would remove them entirely